### PR TITLE
Support client minimization via X11 communication

### PR DESCRIPTION
**Pull request by t-wissmann**
This implements the minimization of windows as specified by ICCCM and
updates the respective X properties as specified by EWMH.

### CODE REVIEW

**Comment by t-wissmann** [comment]
This includes the changes of #1032 and has to be merged after #1032.

File: tests/test_ewmh.py [hunk 1]
```diff
@@ -348,3 +349,44 @@ def test_ewmh_make_client_urgent_no_focus_stealing(hlwm, hc_idle, x11):
...
+    assert (hidden in x11.ewmh.getWmState(winHandle, str=True)) == minimized
+
+
+def xiconifywindow(display, window, screen):
+    """Implementation of XIconifyWindow()"""
+    wm_change_state = display.get_atom('WM_CHANGE_STATE')
+    event = Xlib.protocol.event.ClientMessage(
+        window=window,
+        client_type=wm_change_state,
+        data=(32, [3, 0, 0, 0, 0]))
```
Comment (blarz) [inline tests/test_ewmh.py:370]: The 3 represents the IconicState, right?

Can you maybe put a variable with a more speaking name in there, so this has a less "magic number" feeling, e.g.

iconicstate = 3
...
data=(32, [iconicstate, 0, 0, 0, 0]))

**Review by blarz — CHANGES_REQUESTED** [review]
Just a minor request, otherwise this looks fine to me

File: tests/test_ewmh.py (hunk 1 above)
Comment (t-wissmann) [reply tests/test_ewmh.py]: Right, I've fixed it!

Comment (blarz) [reply tests/test_ewmh.py]: Thanks!

**Review by blarz — APPROVED** [review]
Nice!

**Pull request merged**